snuba/querylog/factory.py

```diff
@@ -0,0 +1,34 @@
+from snuba.clickhouse.query_dsl.accessors import get_time_range
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.factory import get_dataset_name
+from snuba.query.data_source.projects_finder import ProjectsFinder
+from snuba.query.logical import Query
+from snuba.querylog.query_metadata import SnubaQueryMetadata
+from snuba.request import Request
+from snuba.utils.metrics.timer import Timer
+
+
+def get_snuba_query_metadata(
```

Comment:
@evanh No they aren't used outside of validation.py. Initially I put them in the same file as the constructor for their object because I thought since they are both used for object creation they should be located together. But then I saw somewhere else that factory.py file was used for factory function, so I made a factory.py to put it in.